### ISSUE TITLE
ace-link-gnus; possible error in emacs 27

### ISSUE THREAD

**Issue by poulpoulsen** [abo-abo/ace-link#56]
Hello,
i use ace-link-gnus for a while inside emacs 27 and it works.
But until last week it doesn't anymore.
If command is executed, it complains "widget-move: No buttons or fields found"
and nothing happens.

I read an info in the gnus-mailing-list about internal changes. Is this the reason?
Regards
Poul

From:  Lars Ingebrigtsen <larsi@gnus.org>
Subject: Widgets -> Buttons
To: info-gnus-english@gnu.org
Date: Tue, 30 Jul 2019 15:29:26 +0200 (5 days, 3 hours, 47 minutes ago)
Organization: Programmerer Ingebrigtsen
Mail-Followup-To: info-gnus-english@gnu.org
Mail-Copies-To: never

Just a heads up: I've converted the Article buffer in Gnus from using
widgets.el widgets to use button.el buttons.  There should be no
user-visible changes, but it's a pretty fundamental change.  If you have
code that relies on the Article buffer to widgetisised (that's a word)
there may be compatibility problems, and we should try to mitigate those
(if they exist).

The reason for this change is that the widget code is rather opinionated
about how things should work, and some extensions to button behaviour I
wanted to add would be very awkward.  In the button.el paradigm, though,
things are much freer, and I think it should be easy to do these things
now.

-- 
(domestic pets only, the antidote for overdose, milk.)
   bloggy blog: http://lars.ingebrigtsen.no

_______________________________________________
info-gnus-english mailing list
info-gnus-english@gnu.org
https://lists.gnu.org/mailman/listinfo/info-gnus-english

**Comment by abo-abo** [comment]
Thanks for the report. Looks like we will have to extend `ace-link-gnus` to test `emacs-version`.

**Comment by poulpoulsen** [comment]
Hello,
ok that sounds good.
Any change that this will be fixed with checking the emacs version?
I know a stupid question, but i'm not a developer...
Or is there any other solution? I can't find any. Your package was the best.

Regards
Poul

**Comment by abo-abo** [comment]
Looks like `ace-link-help` deals with buttons. Does that work for you in the new gnus?

**Comment by poulpoulsen** [comment]
Hello,
unfortunately not.
Emacs complains "button-label: Wrong type argument: number-or-marker-p,
nil"

I did "M-x ace-link-help" in Article-buffer.
Is this correct?

Regards
Poul

**Comment by abo-abo** [comment]
Yes, it's correct. Too bad it does not work. I'll have a look when I have some time.

**Comment by poulpoulsen** [comment]
Thank you for your work.
I will wait and see.
